### ISSUE TITLE
Add transaction hash in tx related API endpoints (tx Objects)

### ISSUE THREAD

**Issue by dadaphl** [aeternity/protocol#25]
Scenarios
- use tx-hash as primary key
- use tx-hash in urls

Affect api endpoints (so far)
```
GET /block/tx/height/{height}/{tx_index}
GET /block/txs/list/height
GET /block/tx/hash/{hash}/{tx_index}
GET /block/tx/latest/{tx_index}
GET /block/txs/list/hash
```

**Comment by dadaphl** [comment]
height is ambiguous because of forking, not sure if it should be included.

**Comment by velzevur** [comment]
Implemented in aeternity/epoch#686
